src/data/nat/gcd/basic.lean

```diff
@@ -487,27 +487,38 @@ lemma coprime.eq_of_mul_eq_zero {m n : ℕ} (h : m.coprime n) (hmn : m * n = 0)
 def prod_dvd_and_dvd_of_dvd_prod {m n k : ℕ} (H : k ∣ m * n) :
   { d : {m' // m' ∣ m} × {n' // n' ∣ n} // k = d.1 * d.2 } :=
 begin
-cases h0 : (gcd k m),
-case nat.zero
-{ obtain rfl : k = 0 := eq_zero_of_gcd_eq_zero_left h0,
-  obtain rfl : m = 0 := eq_zero_of_gcd_eq_zero_right h0,
-  exact ⟨⟨⟨0, dvd_refl 0⟩, ⟨n, dvd_refl n⟩⟩, (zero_mul n).symm⟩ },
-case nat.succ : tmp
-{ have hpos : 0 < gcd k m := h0.symm ▸ nat.zero_lt_succ _; clear h0 tmp,
-  have hd : gcd k m * (k / gcd k m) = k := (nat.mul_div_cancel' (gcd_dvd_left k m)),
-  refine ⟨⟨⟨gcd k m,  gcd_dvd_right k m⟩, ⟨k / gcd k m, _⟩⟩, hd.symm⟩,
-  apply dvd_of_mul_dvd_mul_left hpos,
-  rw [hd, ← gcd_mul_right],
-  exact dvd_gcd (dvd_mul_right _ _) H }
+  cases h0 : (gcd k m),
+  case nat.zero
+  { obtain rfl : k = 0 := eq_zero_of_gcd_eq_zero_left h0,
+    obtain rfl : m = 0 := eq_zero_of_gcd_eq_zero_right h0,
+    exact ⟨⟨⟨0, dvd_refl 0⟩, ⟨n, dvd_refl n⟩⟩, (zero_mul n).symm⟩ },
+  case nat.succ : tmp
+  { have hpos : 0 < gcd k m := h0.symm ▸ nat.zero_lt_succ _; clear h0 tmp,
+    have hd : gcd k m * (k / gcd k m) = k := (nat.mul_div_cancel' (gcd_dvd_left k m)),
+    refine ⟨⟨⟨gcd k m,  gcd_dvd_right k m⟩, ⟨k / gcd k m, _⟩⟩, hd.symm⟩,
+    apply dvd_of_mul_dvd_mul_left hpos,
+    rw [hd, ← gcd_mul_right],
+    exact dvd_gcd (dvd_mul_right _ _) H }
+end
+
+lemma dvd_mul {x m n : ℕ} :
+  x ∣ (m * n) ↔ ∃ y z, y ∣ m ∧ z ∣ n ∧ y * z = x :=
```

Comment:
Notice that this property is what characterizes "[decomposition monoids](https://d-nb.info/1192415523/34)" and [pre-Schreier domains](https://en.wikipedia.org/wiki/Schreier_domain), which mathlib doesn't currently have. From p. 3 of the first link: "a monoid H is called a decomposition monoid if every element a ∈ H is primal, that is, for every b, c ∈ H such that a | bc there exist a1, a2 ∈ H such that a = a1*a2, a1 | b and a2 | c. A domain R is pre-Schreier if the multiplicative monoid R \ {0} is a decomposition monoid. The notion of a pre-Schreier domain was introduced by Zafrullah in [24] ...

Reply:
Update: I just found[exists_dvd_and_dvd_of_dvd_mul](https://leanprover-community.github.io/mathlib_docs/algebra/gcd_monoid/basic.html#exists_dvd_and_dvd_of_dvd_mul) showing a gcd_monoid is also a decomposition monoid.

Reply:
Nice find! Perhaps I should promote that lemma to an iff in this PR.

Reply:
I've linked together the docstrings a little.